ToolBox/Polygon/PolygonDataQueueHandler.cs

```diff
@@ -507,105 +508,235 @@ private IEnumerable<Tick> GetCryptoTradeTicks(HistoryRequest request)
             }
         }
 
-
         private IEnumerable<Tick> GetEquityQuoteTicks(HistoryRequest request)
         {
             // https://api.polygon.io/v2/ticks/stocks/nbbo/SPY/2020-08-24?apiKey=
 
+            bool isException = false;
+
             var start = request.StartTimeUtc;
             var end = request.EndTimeUtc;
+            var offsetStart = Time.DateTimeToUnixTimeStampNanoseconds(start);
+            var offsetEnd = Time.DateTimeToUnixTimeStampNanoseconds(end);
 
-            while (start <= end)
+            // Continues until offsetStart gets to the end of the set time at offsetEnd
+            while (offsetStart < offsetEnd)
             {
+                // Skips Saturday and Sunday
+                if (!start.IsCommonBusinessDay())
+                {
+#if DEBUG
+                    Log.Trace("Weekends are not processed " + start.ToString(System.Globalization.CultureInfo.GetCultureInfo("en-US")));
+#endif
+                    start = start.Date.AddDays(1);
+                    break;
```

Comment:
The offset is used in the Pagination for the API and this design pattern is specific to the Polygon API behavior.   This code iterates through each day at a time and the start offset is only for the pagination.

You can see from the REST endpoint that it is based on a day.    https://polygon.io/docs/get_v2_ticks_stocks_trades__ticker___date__anchor

This is one of the problems of the original code that has been fixed with this revision.

Reply:
Right, my point was that as is I don't think it's behaving as expected, `break` will just cause it to break the while loop and return, and the result of `start = start.Date.AddDays(1)` isn't being used?

Reply:
I'll take another look.   The loop offset iterates over 1 day at a time.     So, if the day is not a common business day, then the offset loop stops and the outer loop should process the next day.

Reply:
This isn't behaving as expected, the `break` will exit the `while (offsetStart < offsetEnd)`

Reply:
@Martin-Molinero     This is the expected behavior.   The Polygon API is setup for processing only one day at a time.   When this breaks, the calling method returns and processes the next day, which resets the offsetStart and offsetEnd.

downloader.Get(symbol, resolution, utcStartDate.AddDays(i), utcEndDate.AddDays(i + 1), tickType)

Reply:
That might work for your application @TightLinesOfCode that is not the API exposed in this method; it is expected the method would return a complete history request with all the days. That looping and selection of the active days should be done internally. 

We should not rely on the caller's application of the interface to implement it or it will not be portable to other applications.

Reply:
PS: Ideally this should be using the Market Hours DB; what if Monday is a holiday; we can look ahead and remove this loop.